### PR TITLE
removed unnecessary simplification in solveset

### LINKED ISSUE #14300: Neither solve nor solveset can solve 1 - exp(-18000000*x) = y for x
```
var('x y')
solve(1 - exp(-18000000*x) - y, x)   # hangs
solveset(1 - exp(-18000000*x) - y, x)  # hangs
```
At least in my case (Ubuntu), the entire OS hangs, not just SymPy.

Based on a [Stack Overflow post](https://stackoverflow.com/questions/48903574/how-to-lambdify-an-intersection)

### PR DESCRIPTION
**Pull request by Yathartha22**
#### References to other Issues or PRs
fixes: #14300 

#### Brief description of what is fixed or changed

It seems that `1 - exp(-18000000*x) - y` was simplified to `(-y*exp(18000000*x) + exp(18000000*x) - 1)*exp(-18000000*x)` and was further solved as a polynomial. Although such equations can be easily handled by `_invert`.

#### Other comments

Currently, all rationals are combined using together for ease of calculation, but this step could sometime make the equations more complex to solve, as a result this should be done wherever necessary.

### CODE REVIEW

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
 
 from types import GeneratorType
 
+from sympy import factor
```
Comment (Yathartha22) [inline sympy/solvers/solveset.py]: this got here by mistake, will be removing in the next commit.

**Comment by Yathartha22** [comment]
There is [this](https://github.com/sympy/sympy/blob/master/sympy/solvers/tests/test_solveset.py#L612) one test case that is failing. It seems that the recent changes in solveset might be the issue because similar changes when done in `1.1.1` would pass the case.
ping @smichr

File: sympy/solvers/solveset.py [hunk 2]
```diff
@@ -790,7 +791,7 @@ def _solveset(f, symbol, domain, _check=False):
     from sympy.simplify.simplify import signsimp
 
     orig_f = f
-    tf = f = together(f)
+    tf = together(f)
     if f.is_Mul:
```
Comment (smichr) [inline sympy/solvers/solveset.py]: This should now be 

```
if tf.is_Mul:
    coeff, f = tf.as_independent(symbol, as_Add=False)
    if coeff...
        f = tf
```

Comment (Yathartha22) [reply sympy/solvers/solveset.py]: By doing this we are in a way doing the same thing. We are checking the combined form of the eq (`(-y*exp(18000000*x) + exp(18000000*x) - 1)*exp(-18000000*x)`).

Comment (smichr) [reply sympy/solvers/solveset.py]: ok, then let's remove the tf calculation and just write `f = together(f)` in the `if coeff...` block.

Comment (smichr) [reply sympy/solvers/solveset.py]: This takes care of the test, too:

```diff
diff --git a/sympy/solvers/solveset.py b/sympy/solvers/solveset.py
index ce6074f..693acff 100644
--- a/sympy/solvers/solveset.py
+++ b/sympy/solvers/solveset.py
@@ -685,8 +685,8 @@ def _solve_abs(f, symbol, domain):
             complex domain.'''))
     p, q, r = Wild('p'), Wild('q'), Wild('r')
     pattern_match = f.match(p*Abs(q) + r) or {}
-    if not pattern_match.get(p, S.Zero).is_zero:
-        f_p, f_q, f_r = pattern_match[p], pattern_match[q], pattern_match[r]
+    f_p, f_q, f_r = [pattern_match.get(i, S.Zero) for i in (p, q, r)]
+    if not (f_p.is_zero or f_q.is_zero):

         domain = continuous_domain(f_q, symbol, domain)
         q_pos_cond = solve_univariate_inequality(f_q >= 0, symbol,
@@ -787,12 +787,12 @@ def _solveset(f, symbol, domain, _check=False):
     from sympy.simplify.simplify import signsimp

     orig_f = f
-    tf = f = together(f)
     if f.is_Mul:
         coeff, f = f.as_independent(symbol, as_Add=False)
         if coeff in set([S.ComplexInfinity, S.NegativeInfinity, S.Infinity]):
-            f = tf
-    if f.is_Add:
+            f = together(f)
+    elif f.is_Add:
         a, h = f.as_independent(symbol)
         m, h = h.as_independent(symbol, as_Add=False)
         if m not in set([S.ComplexInfinity, S.Zero, S.Infinity,

```

**Pull request merged**